### PR TITLE
Added solution

### PR DESCRIPTION
**Pull request by AndrewPolishchuc**

### CODE REVIEW

File: src/main/java/core/basesyntax/controller/ConsoleHandler.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.imp.BetDaoImpl;
+import core.basesyntax.model.Bet;
+import java.util.Scanner;
+
+public class ConsoleHandler {
+    private BetDao betDao = new BetDaoImpl();
+
+    public ConsoleHandler() {
+
```
Comment (andromash) [inline src/main/java/core/basesyntax/controller/ConsoleHandler.java]:
```suggestion
```

File: src/main/java/core/basesyntax/controller/ConsoleHandler.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    public ConsoleHandler(BetDao betDao) {
+        this.betDao = betDao;
+    }
```
Comment (andromash) [inline src/main/java/core/basesyntax/controller/ConsoleHandler.java]: there is no need in this constructor

File: src/main/java/core/basesyntax/controller/ConsoleHandler.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        while (true) {
+            String command = scanner.nextLine();
+            if (command.equalsIgnoreCase("q")) {
+                return;
+            }
+            Bet bet = null;
+            try {
+                String[] betData = command.split(" ");
+                int value = Integer.parseInt(betData[0]);
+                double risk = Double.parseDouble(betData[1]);
```
Comment (andromash) [inline src/main/java/core/basesyntax/controller/ConsoleHandler.java:51]: probably we need to read the user as well and put it to storage

**Review by andromash — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/controller/ConsoleHandler.java (hunk 1 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/controller/ConsoleHandler.java]: Done

File: src/main/java/core/basesyntax/controller/ConsoleHandler.java (hunk 3 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/controller/ConsoleHandler.java:51]: I agree

File: src/main/java/core/basesyntax/controller/ConsoleHandler.java (hunk 2 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/controller/ConsoleHandler.java]: You're right

File: src/main/java/core/basesyntax/controller/ConsoleHandler.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                String name = userData[0];
+                String password = userData[1];
+                user = new User(name, password);
+            } catch (NumberFormatException | IndexOutOfBoundsException e) {
+                System.out.println("The entered data is incorrect");
+                continue;
+            }
+            userDao.add(user);
+            System.out.println(user);
+            while (true) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/controller/ConsoleHandler.java]: let's not create a loop inside a loop
try passing everything you need for both models in one line

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/core/basesyntax/factory/Factory.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import core.basesyntax.exceptions.MissingAnnotationException;
+import core.basesyntax.lib.Dao;
+
+public class Factory {
+    private static BetDao betDao;
+    private static UserDao userDao;
+
+    public static BetDao getBetDaoImpl() {
+        if (betDao == null) {
+            if (BetDaoImpl.class.getAnnotation(Dao.class) == null) {
```
Comment (andromash) [inline src/main/java/core/basesyntax/factory/Factory.java]: this check is not related to the Factory pattern. Move it to Injector

File: src/main/java/core/basesyntax/factory/Factory.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+                throw new MissingAnnotationException("Missing annotation");
+            }
+            return new BetDaoImpl();
+        }
+        return betDao;
+    }
+
+    public static UserDao getUserDaoImpl() {
+        if (userDao == null) {
+            if (UserDaoImpl.class.getAnnotation(Dao.class) == null) {
```
Comment (andromash) [inline src/main/java/core/basesyntax/factory/Factory.java]: same here

**Review by andromash — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/factory/Factory.java (hunk 5 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/factory/Factory.java]: Agree

File: src/main/java/core/basesyntax/factory/Factory.java (hunk 6 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/factory/Factory.java]: Done

File: src/main/java/core/basesyntax/controller/ConsoleHandler.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.lib.Inject;
+import core.basesyntax.model.Bet;
+import core.basesyntax.model.User;
+import java.util.Scanner;
+
+public class ConsoleHandler {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/controller/ConsoleHandler.java:12]:
```suggestion
```

File: src/main/java/core/basesyntax/factory/Factory.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+public class Factory {
+    private static final Factory factory = new Factory();
+
+    private Factory() {
+
+    }
+
+    private Factory getInstance() {
+        return factory;
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/factory/Factory.java]: why have decided to make Factory constructor private?

**Review by dmytro-hryhoruk — COMMENTED** [review]
I don't see any changes related to Injector in this PR after the last commit

File: src/main/java/core/basesyntax/factory/Factory.java (hunk 8 above)
Comment (AndrewPolishchuc) [reply src/main/java/core/basesyntax/factory/Factory.java]: Implementation singleton
